### ISSUE TITLE
Postgres money type breaks visualization (no avg(money))

### ISSUE THREAD

**Issue by philippe-lavoie** [metabase/metabase#11498]
**Describe the bug**
I have a table with a money type and in the x-ray view, the visual component says there is an error.

**Logs**

12-11 14:24:35 [1mWARN middleware.process-userland-query[0m :: Query failure {:status :failed,
 :class org.postgresql.util.PSQLException,
 :error "ERROR: function avg(money) does not exist\n  Hint: No function matches the given name and argument types. You might need to add explicit type casts.\n  Position: 259",
 :stacktrace

**To Reproduce**
Create a table with some money type in it. Add a row or two (I guess). Use metabase.

**Expected behavior**
Should not give an error

**Severity**
Critical, I had to change the type to numeric.

**Additional context**
From the log, doing a 'avg(money::numeric)' would probably solve it.

**Comment by flamber** [comment]
Hi @philippe-lavoie Can you post "Diagnostic Info" from Admin > Troubleshooting. Related to #3754?

**Comment by camsaul** [comment]
Is this a P1?

**Comment by camsaul** [comment]
@mazameli curious why this is a P1. It looks like a duplicate of #3754 which was a P3

**Comment by mazameli** [comment]
@camsaul Fair point. Doesn't seem to be too widespread, so I'm downgrading it to P2. I'm guessing I labeled this one as P1 since it appeared to break visualizations vs. #3754.

**Comment by camsaul** [comment]
This is actually not allowed in Postgres, here's the output from `psql`:

```sql
money_columns_test=# SELECT AVG(price) FROM bird_prices;
ERROR:  function avg(money) does not exist
```
 
I should be able to work around it however

**Comment by camsaul** [comment]
Fixed by #11717
